python/cudf/cudf/_lib/json.pyx

```diff
@@ -76,12 +76,22 @@ cpdef read_json(object filepath_or_buffer,
         if isinstance(dtype, abc.Mapping):
             c_dtypes.reserve(len(dtype))
             for k, v in dtype.items():
+                if cudf.utils.dtypes.is_categorical_dtype(v):
+                    raise NotImplementedError(
+                        "CategoricalDtype as dtype is not yet "
+                        "supported in JSON reader"
+                    )
```

Comment:
Should this check be in libcudf, so they can also apply to the C++ API users?

Reply:
Seems like we an error throw from c++ layer which will say `"unknow type_id.."`, and would propagate as a `RunTimeError` into python layer. To make it look nicer I had to add a check at python layer.